pkg/apiserver/rest/scan_controller.go

```diff
@@ -98,6 +109,27 @@ func (s *ServerImpl) PatchScansScanID(ctx echo.Context, scanID models.ScanID, pa
 		return sendError(ctx, http.StatusBadRequest, fmt.Sprintf("failed to bind request: %v", err))
 	}
 
+	// check that an asset with that id exists
```

Comment:
The comment here should state `// check that a scan with that id exists`.

Reply:
Actually, do we need this check? I think we only need to move the valid transition check code block down to line 139, right?

Reply:
_check **if** a scan with that id exists_